opencensus/proto/stats/metrics/metrics.proto

```diff
@@ -215,12 +216,27 @@ message DistributionValue {
     // The number of values in each bucket of the histogram, as described in
     // bucket_bounds.
     int64 count = 1;
-
-    // TODO: Add support for exemplars.
   }
 
   // If the distribution does not have a histogram, then omit this field.
   // If there is a histogram, then the sum of the values in the Bucket counts
   // must equal the value in the count field of the distribution.
   repeated Bucket buckets = 6;
+
+  // Exemplars are example points that may be used to annotate aggregated
+  // Distribution values. They are metadata that gives information about a
+  // particular value added to a Distribution bucket.
+  message Exemplar {
+    // Value of the exemplar point.
+    double value = 1;
```

Comment:
Yes the exemplar value field is required because it determines which bucket the exemplar belongs to, while things in 'attachments' are optional.

Reply:
> Is the value useful for reasons other than associating it with the bucket (I would expect so)?

I think that depends on the backend. For Stackdriver, they didn't mention anything specific except that [value is used to determine to which bucket the exemplar belongs.](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries#exemplar).

> Also, should Exemplar be a component of Bucket?

We can do that. I added Exemplar as a separate repeated field because this matches the [proto definition on Stackdriver](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries#distribution). In our definition Bucket can have more information than just bucket count, and that's different from Stackdriver.

Reply:
Reverted the change on having Exemplar as a part of Bucket, per our discussion today.